app/scripts/controllers/push-platform-notifications/services/services.ts

```diff
@@ -183,27 +183,29 @@ export async function activatePushNotifications(
 
 export async function listenToPushNotifications(
   onNewNotification: (notification: Notification) => void,
-): Promise<Unsubscribe> {
+  onNotificationClicked: (notification: Notification) => void,
+): Promise<() => void> {
   // Firebase
   const messaging = await getFirebaseMessaging();
-  const unsubscribe = onBackgroundMessage(
```

Comment:
In the file above, there is a rename `firebaseUnsubscribe` -> `pushListenerUnsubscribe`. Here there is a rename `unsubscribe` -> `unsubscribeFirebase`. What is the difference between these two contexts that lead to the different names?

Reply:
I agree - I will rename this to `pushListenerUnsubscribe`.

In this context, when listening to push notifications we need to listen to 2 things:
1. When a push notification is received (and how we need to handle it to show to users).
2. When a push notification is clicked.

Since we have 2 listeners, we need to make sure we unsubscribe from both if they need to be torn down.